vignettes/markdown.Rmd

```diff
@@ -141,33 +148,52 @@ URLs inside angle brackets are also automatically converted to hyperlinks:
-   same package: `[object]`. These links that *not* typeset as code,
-   so if you want them as code, enclose them in backticks (inside the
-   brackets).
-3. Link to a function from another package: `[pkg::func()]`. These links
-   will be typeset as code.
-4. Link to a (non-function) object in another package: `[pkg::object]`.
-   These links will not be typeset as code.
-5. Link to an object in the same package, with a different link text:
-   `[link text][object]`. Here `object` can be a function, but the link
-   text is not typeset as code.
-6. Link to an object in another package, with different link text:
-   `[link text][pkg::object]`. This is not typeset as code.
-
-S3/S4 classes can be linked the same way:
-
-```r
-#' * [terms][terms.object] becomes \link[=terms.object]{terms}
-#' * [abc][abc-class] becomes \link[=abc-class]{abc}
-```
-
-Or a shorthand notation can be used: `[abc-class]` is converted to `\linkS4class{abc}` and `[pkg::abc-class]` is converted to `\link[pkg:abc-class]{pkg::abc}`.
+Markdown notation can also be used to create links to other help topics. There are two basic forms:
+
+  * `[ref]`: The target topic and the link text are the same.
+  * `[text][ref]`: Link text differs from the target.
+
+with variations depending on whether `ref` is in the same package and whether the link is automatically typeset as code. Use backticks inside the square brackets to typeset the link text as code, when it would not be otherwise:
+
+  * ``[`ref`]``
+  * ``[`text`][ref]``
+
++-----------------------+------------------------+--------------------------+
+| Examples              | Help topic\            | Notes                    |
+|                       | for what?              |                          |
++=======================+========================+==========================+
+| `[func()]`            | a function in\         | Always typeset as code.\            |
+|                       | same package           | In Rd: `\code{\link[=func]{func()}` | 
++-----------------------+------------------------+--------------------------+  
+| `[object]`\           | a topic in\            | Use for a topic where you documented\ |
+| ``[`object`]``        | same package           | `NULL` and set its name via `@name`,\ |
+|                       |                        | e.g., a class, dataset, or theme.\    | 
+|                       |                        | Bring your own backticks.\            |
+|                       |                        | In Rd: `\link{object}`                |
++-----------------------+------------------------+--------------------------+
+| `[text][func()]`\     | a function or topic\   | Use when you want to provide your\               |
+| `[text][object]`\     | in same package        | own link text.\                                  |
+| `[abc][abc-class]`    |                        | Bring your own backticks.\                       |
```

Comment:
For `text`  you usually don't want backticks, that is just text. So maybe the `abc-class` case could  be on its own row?